tools/airgap_linter.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+#
+# Validates that the supplied framework (identified by the directory passed in) will function, to the best
+# of our knowledge, in an airgapped cluster.
+#
+# The checks are:
+#   - No external URIs defined in anything but resource.json
+#   - No images defined in anything but resource.json (and templated into the svc.yml)
+#
+
+import re
+import sys
+import os
+
+
+def extract_uris(file_name):
+    with open(file_name, "r") as file:
+        lines = file.readlines()
+
+    matcher = re.compile(".*(https:\/\/|http:\/\/)([^\/]*)", re.IGNORECASE)
+    matches = []
+    for line in lines:
+        match = matcher.match(line)
+        if match:
+            matches.append(match.group(2))
+
+    return matches
+
+
+def validate_uri(uri):
+    # URIs are valid if they:
+    #   - end with .mesos
+    #   - contain .dcos
+    if ".dcos" in uri:
+        return True
+
+    if uri.endswith(".mesos"):
+        return True
+
+    return False
+
+
+
+def validate_uris_in(file_name):
```

Comment:
feels like either this should be `validate_uris`, or `validate_all_uris` should be `validate_all_uris_in`